src/Featurizers/CountVectorizerFeaturizer.h

```diff
@@ -0,0 +1,481 @@
+    _regex(std::move(regex)),
+    _ngram_min(std::move(ngram_min)),
+    _ngram_max(std::move(ngram_max)),
+    _parse_func([this](void) -> ParseFunctionType {
+            //initialize parse function
+            ParseFunctionType parseFunc;
+            if (_analyzer == AnalyzerMethod::Word) {
+                if (!_regex.empty()) {
+                    parseFunc = [this] (std::string const & input, std::function<void (StringIterator, StringIterator)> const &callback) {
+                        Microsoft::Featurizer::Strings::ParseRegex<std::string::const_iterator, std::regex>(
+                            input,
+                            std::regex(_regex),
+                            callback
+                        );
+                    };
+                } else if (_ngram_min == 1 && _ngram_max == 1) {
+                    parseFunc = [] (std::string const & input, std::function<void (StringIterator, StringIterator)> const &callback) {
+                        Microsoft::Featurizer::Strings::Parse<std::string::const_iterator, std::function<bool (char)>>(
+                            input,
+                            [] (char c) {return std::isspace(c);},
+                            callback
+                        );
+                    };
+                } else {
+                    parseFunc = [this] (std::string const & input, std::function<void (StringIterator, StringIterator)> const &callback) {
+                        Microsoft::Featurizer::Strings::ParseNgramWord<std::string::const_iterator, std::function<bool (char)>>(
+                            input,
+                            [] (char c) {return std::isspace(c);},
+                            _ngram_min,
+                            _ngram_max,
+                            callback
+                        );
+                    };
+                }
+            } else if (_analyzer == AnalyzerMethod::Char) {
+                parseFunc = [this] (std::string const & input, std::function<void (StringIterator, StringIterator)> const &callback) {
+                    Microsoft::Featurizer::Strings::ParseNgramChar<std::string::const_iterator>(
+                        input,
+                        _ngram_min,
+                        _ngram_max,
+                        callback
+                    );
+                };
+            } else {
+                assert(_analyzer == AnalyzerMethod::Charwb);
+
+                parseFunc = [this] (std::string const & input, std::function<void (StringIterator, StringIterator)> const &callback) {
+                    Microsoft::Featurizer::Strings::ParseNgramCharwb<std::string::const_iterator, std::function<bool (char)>>(
+                        input,
+                        [] (char c) {return std::isspace(c);},
+                        _ngram_min,
+                        _ngram_max,
+                        callback
+                    );
+                };
+            }
+            return parseFunc;

```

Comment:
I didn't expect to see this code here - isn't this handled by DocumentStatisticsEstimator?